File: 0075-sort-colors/0075-sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        //DUTCH NATIONAL FLAG ALGORITHM
        int n =  nums.size() - 1;
        int low = 0, mid = 0, high = n;
        while(mid <= high) {
            if(nums[mid] == 0 ) {
                swap( nums[low],nums[mid]);
                mid++;
                low++;
            } else if(nums[mid] == 1) {
                mid++;
            } else {
                swap(nums[high],nums[mid]);
                high--;
            }
        }
        
    }
};

File: 0075-sort-colors/0075-sort-colors.java
class Solution {
    public void sortColors(int[] nums) {
        int n = nums.length;
        int temp[] = {0,0,0};
        for(int i = 0 ; i < n; i++) {
            temp[nums[i]]++;
        }
        int i = 0;
        int j = 0;
        while(j < n) {
            if(temp[i] > 0) {
                nums[j] = i;
                j++;
                temp[i]--;
            }else {
                i++;
            }
        }
    }
}